### DISCUSSION TITLE
modbusrtu - communication with five clients -  how to?

### DISCUSSION THREAD

**Discussion by rozrabiak** [eModbus/eModbus#257] [eModbus applications]
Hi, how to communicate with five clients from modbusrtu using this library?
I need one ModbusRTUMaster (server) and five Arduino Modbus Clients.
Is it possible?

**Comment by Miq1** [comment]
Let me sort that out: any RTU Modbus legally can have only one and just one client. The client has no ID as such, so f.i. a server would not know which client was requesting.

If you have one device providing data and five devices need to read that data, you should consider different approaches. Some ideas:

- use Modbus TCP via WIFI instead. Here a server may answer any request regardless of the number of clients requesting
- if you need to stick to RTU, turn around the roles and let the device providing the data the client. Then use the broadcast request to cyclically send the data to whatever devices are listening as servers on the bus. Intentionally the servers are not allowed to respond to a broadcast and have no influence on when the client will send its data
- use another protocol than Modbus that allows multiple clients on a serial bus. Note that this will require a different wiring in most cases

**Comment by bertmelis** [comment]
Mind also:
master = client
slave = server
